### PR TITLE
Set up Storybook📚

### PR DESCRIPTION
**Pull request by ryotogashi**

### CODE REVIEW

**Comment by ryotogashi** [comment]
#4

**Comment by ryotogashi** [comment]
@axross Can you check?

I changed 
tsconfig.json
```diff
- # "jsx": "preserve"
+ # "jsx": "reaxt"
```

I couldn't find a way to write compilerOptions in Storybook's config.
You may have a reason set "preserve" value for "jsx" key but I thought "preserve" value means that jsx file won't change to js file. Let me know it was actually your intention.

**Comment by axross** [comment]
@ryotogashi 

`jsx: "preserve"` is required by Next.js because it uses Webpack and Babel in the back. In order to let it work well, the code output from TypeScript compiler is supposed to be non-transpiled JSX.

File: pages/_document.tsx [hunk 1]
```diff
@@ -1,12 +1,6 @@
...
-  DocumentInitialProps,
-  Html,
-  Head,
-  Main,
-  NextScript
-} from "next/document";
-import { ServerStyleSheet } from "styled-components";
+import NextDocument, { DocumentContext, DocumentInitialProps, Html, Head, Main, NextScript } from 'next/document';
+import { ServerStyleSheet } from 'styled-components';
+import * as React from 'react';
```
Comment (axross) [inline pages/_document.tsx]: These changes are not what you intended, they are by your editor's auto-formatting, right?

Seems not related on this pull request. Could you restore these changes?

File: pages/_document.tsx [hunk 2]
```diff
@@ -23,9 +17,7 @@ export default class Document extends NextDocument {
     );
   }
 
-  static async getInitialProps(
-    ctx: DocumentContext
-  ): Promise<DocumentInitialProps> {
+  static async getInitialProps(ctx: DocumentContext): Promise<DocumentInitialProps> {
```
Comment (axross) [inline pages/_document.tsx]: Same here

File: pages/index.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import * as React from 'react';
+import { storiesOf } from '@storybook/react';
+
+import Home from './index';
```
Comment (axross) [inline pages/index.stories.tsx]:
```suggestion
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import Home from './index';
```

File: pages/index.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+
+storiesOf('Home', module).add('home', () => {
+  return <Home />;
+});
```
Comment (axross) [inline pages/index.stories.tsx]: `storiesOf` API will be deprecated. Could you refer https://github.com/axross/react-holdem and use `addon-docs`-based style instead?

File: .storybook/main.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+module.exports = {
+  stories: ['../pages/**/*.stories.tsx'],
+  addons: ['@storybook/addon-storysource'],
```
Comment (axross) [inline .storybook/main.js]: After you start to use `addon-docs`, you will be no need to use `storysource` as well.

File: .storybook/main.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+module.exports = {
+  stories: ['../pages/**/*.stories.tsx'],
```
Comment (axross) [inline .storybook/main.js]: Why did you specify it?

File: package.json [hunk 7]
```diff
@@ -15,15 +15,24 @@
...
+    "@types/storybook__react": "^5.2.1",
     "@types/styled-components": "^5.0.1",
+    "react-docgen-typescript-loader": "^3.6.0",
+    "ts-loader": "^6.2.1",
     "typescript": "^3.8.3"
   },
   "scripts": {
     "dev": "next dev",
     "build": "next build",
+    "build-story": "build-storybook",
```
Comment (axross) [inline package.json]:
```suggestion
```

File: tsconfig.json [hunk 8]
```diff
@@ -3,13 +3,14 @@
     "target": "esnext",
     "module": "esnext",
     "lib": ["dom", "dom.iterable", "esnext"],
-    "jsx": "preserve",
+    "jsx": "react",
```
Comment (axross) [inline tsconfig.json]:
```suggestion
    "jsx": "preserve",
```

ref. https://github.com/agnosticful/chipstackoverflow/pull/7#issuecomment-594218904

File: tsconfig.json [hunk 9]
```diff
@@ -3,13 +3,14 @@
...
     "strict": true,
     "noUnusedLocals": true,
     "noUnusedParameters": true,
     "noImplicitReturns": true,
     "noFallthroughCasesInSwitch": true,
     "moduleResolution": "node",
+    "rootDirs": ["pages"],
```
Comment (axross) [inline tsconfig.json]: Should be no need.

```suggestion
```

**Comment by ryotogashi** [comment]
@axross thank you for your review I fixed them.

File: pages/_document.tsx (hunk 1 above)
Comment (ryotogashi) [reply pages/_document.tsx]:
```diff
+ import * as React from 'react';
```
I needed it for compile error. Is it ok?

Comment (axross) [reply pages/_document.tsx]: Yep, thanks for the change!

File: .storybook/preview-head.html [hunk 10]
```diff
@@ -0,0 +1,4 @@
+<link
+  href="https://fonts.googleapis.com/css?family=Montserrat|Open+Sans|Oxanium|Varela+Round|Work+Sans&display=swap"
+  rel="stylesheet"
+/>
```
Comment (axross) [inline .storybook/preview-head.html]: We don't need this.

File: __tests__/pages/index.spec.tsx [hunk 11]
```diff
@@ -1,9 +1,9 @@
...
-import * as TestRenderer from 'react-test-renderer';
-import Home from '../../pages/index';
+import * as React from "react";
+import * as TestRenderer from "react-test-renderer";
+import Home from "../../pages/index";
 
-describe('Home test', () => {
-  it('tests home page render correctly', () => {
+describe("Home test", () => {
+  it("tests home page render correctly", () => {
```
Comment (axross) [inline __tests__/pages/index.spec.tsx]: These are effected by Prettier, right? Supposed not to change in this PR.

File: pages/index.stories.mdx [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import { Description, Meta, Preview } from '@storybook/addon-docs/blocks';
+import Home from './index';
+
+<Meta title="Home" component={Home} />
+
+# Home
+
+<Preview>
+  <Home />
+</Preview>
```
Comment (axross) [inline pages/index.stories.mdx]: Nice! ✨

File: __tests__/pages/index.spec.tsx (hunk 11 above)
Comment (ryotogashi) [reply __tests__/pages/index.spec.tsx]: Yes, I thought I am supposed to format before the merge, no?

Comment (axross) [reply __tests__/pages/index.spec.tsx]: If it is, it is supposed to be formatted in Prettier's PR, right?

Every PR should have a single responsibility. In order to format this file, we should create another PR that has responsibility to format it.

Comment (ryotogashi) [reply __tests__/pages/index.spec.tsx]: I understand! thank you.
